modules/subscriptions.php

```diff
@@ -667,25 +667,31 @@ function widget( $args, $instance ) {
 			</form>
 
 			<script>
-				( function( d ) {
-					if ( ( 'placeholder' in d.createElement( 'input' ) ) ) {
-						var label = d.getElementById( 'jetpack-subscribe-label' );
-	 					label.style.clip 	 = 'rect(1px, 1px, 1px, 1px)';
-	 					label.style.position = 'absolute';
-	 					label.style.height   = '1px';
-	 					label.style.width    = '1px';
-	 					label.style.overflow = 'hidden';
-					}
-				} ) ( document );
-
-				// Special check for required email input because Safari doesn't support HTML5 "required"
-				jQuery( '#subscribe-blog-<?php echo $widget_id; ?>' ).submit( function( event ) {
-					var requiredInput = jQuery( this ).find( '.required' );
-					if ( requiredInput.val() == '' ) {
-						event.preventDefault();
-						requiredInput.focus();
-					}
-				});
+			( function( d ) {
+				if ( ( 'placeholder' in d.createElement( 'input' ) ) ) {
+					var label = d.getElementById( 'jetpack-subscribe-label' );
+						label.style.clip 	 = 'rect(1px, 1px, 1px, 1px)';
+						label.style.position = 'absolute';
+						label.style.height   = '1px';
+						label.style.width    = '1px';
+						label.style.overflow = 'hidden';
+				}
+			
+				var form = d.getElementById('subscribe-blog-<?php echo $widget_id; ?>'), 
+					input = d.getElementById('<?php echo esc_attr( $subscribe_field_id ); ?>'),
+					handler = function(event) {
+						if (input.value === '') {
+							input.focus();
+							return false; 
```

Comment:
I have tested this change, and while it works fine in Firefox and IE, it doesn't look like `return false` works for Safari. I suggest checking for `event.preventDefault` and using it instead here.